### PR TITLE
Fix filter by thingId list data operation

### PR DESCRIPTION
**Pull request by esdrasjnr**
What does this implement/fix? Explain your changes.
---------------------------------------------------
This patch fixes the list data operation, allowing it to filter data by the thing id correctly

Where has this been tested?
---------------------------
**Operating System/Platform:** Ubuntu 18.04
**NPM Version:** 12.15
**NodeJS Version:** 6.14

[Test script with KNoT SDK](https://codeshare.io/240jE7)

### CODE REVIEW

**Review by Igoraamc — APPROVED** [review]
LGTM! 🥳

File: pkg/interactor/list_data.go [hunk 1]
```diff
@@ -10,94 +10,55 @@ import (
 
 // List provides all the thing's data owned by an user
 func (d *DataInteractor) List(token string, query *entities.Query) ([]entities.Data, error) {
-	data, err := d.getDevicesData(token, query)
+	things, err := d.things.List(token)
 	if err != nil {
-		return []entities.Data{}, err
+		return nil, fmt.Errorf("error getting list of things: %w", err)
```
Comment (netoax) [inline pkg/interactor/list_data.go:19]: This error should still describe an authorization failure.

File: pkg/interactor/list_data.go [hunk 2]
```diff
@@ -10,94 +10,55 @@ import (
...
 	}
 
-	data, err = d.filterDataBySensorID(data, query.ThingID, query.SensorID)
-	if err != nil {
-		return []entities.Data{}, err
+	if query.ThingID != "" {
+		things = []*btEntities.Thing{{ID: query.ThingID}}
```
Comment (netoax) [inline pkg/interactor/list_data.go]: Did you test all the endpoints? Notice that we should return all the data from the things whose owner is a single user. That's what `getDevicesData` tried to do.

File: pkg/interactor/list_data.go [hunk 3]
```diff
@@ -10,94 +10,55 @@ import (
...
+			return nil, fmt.Errorf("error getting data: %w", err)
 		}
-
 		data = append(data, thingData...)
 	}
-
 	return data, nil
 }
 
-func (d *DataInteractor) verifyAuthorization(token, id string) error {
```
Comment (netoax) [inline pkg/interactor/list_data.go:64]: Why did you remove this operation? List things by using the passed token isn't enough because we need to verify if the `query.ThingID` matches with a thing owned by that user.

File: pkg/interactor/list_data.go [hunk 4]
```diff
@@ -10,94 +10,55 @@ import (
...
-	things, err := d.things.List(token)
+	data, err := d.getAllData(things, query)
 	if err != nil {
-		return data, fmt.Errorf("%v: %w", ErrValidToken, err)
+		return nil, fmt.Errorf("error getting the data: %w", err)
 	}
 
-	if query.ThingID != "" {
-		err := d.verifyAuthorization(token, query.ThingID)
+	if query.SensorID != "" {
```
Comment (netoax) [inline pkg/interactor/list_data.go:34]: I would prefer not to come back to a less modular code in comparison with the current code. So, if it's necessary to refactor this interactor let's do it in another commit apart from the specific bug fix, and let's keep it as clean as possible.

Comment (netoax) [reply pkg/interactor/list_data.go:34]: Let me know if you have any questions.

**Review by netoax — CHANGES_REQUESTED** [review]
Thanks for the PR. I made some comments in order to help you improve it.

File: pkg/interactor/list_data.go (hunk 1 above)
Comment (esdrasjnr) [reply pkg/interactor/list_data.go:19]: In this case I suppose that the authorization error will come from `List` method. What do you think?

File: pkg/interactor/list_data.go (hunk 2 above)
Comment (esdrasjnr) [reply pkg/interactor/list_data.go]: I have tested with storage SDK with all three methods: `listData`, `listDataByDevice` and `listDataBySensor`. But I gonna make more intensive tests

File: pkg/interactor/list_data.go (hunk 3 above)
Comment (esdrasjnr) [reply pkg/interactor/list_data.go:64]: Yeah, you're right. I've wrongly assumed that `query.ThingID` would always be related to the user token owner. Fixed it!

File: pkg/interactor/list_data.go (hunk 4 above)
Comment (esdrasjnr) [reply pkg/interactor/list_data.go:34]: Part of this refactor was necessary exactly for fix the listing data bug. I've removed some functions since it appeared to be simple to maintain the logic inside the `List` main function, but I could add them back

File: pkg/interactor/list_data.go (hunk 1 above)
Comment (netoax) [reply pkg/interactor/list_data.go:19]: Is this happening?

File: pkg/interactor/list_data.go (hunk 4 above)
Comment (netoax) [reply pkg/interactor/list_data.go:34]: No problem, I will wait for you with respect to the outcome of the tests so that we can decide what to do.

File: pkg/interactor/list_data.go (hunk 2 above)
Comment (netoax) [reply pkg/interactor/list_data.go]: @esdrasjnr I think you can put the results on the PR description so that I can easily analyze.

**Review by netoax — APPROVED** [review]
LGTM!
